build/source/engine/allocspace.f90

```diff
@@ -106,36 +106,71 @@ subroutine allocate_gru_struc(nGRU,nHRU,err,message)
  mode=nf90_noWrite
  call nc_file_open(trim(infile), mode, ncid, err, cmessage)
  if(err/=0)then; message=trim(message)//trim(cmessage); return; endif
-
- ! get gru_ix dimension length
- err = nf90_inq_dimid(ncid, "ngru", gruDimID);             if(err/=0)then; message=trim(message)//'problem finding nGRU dimension'; return; endif
- err = nf90_inquire_dimension(ncid, gruDimID, len = nGRU); if(err/=0)then; message=trim(message)//'problem reading nGRU dimension'; return; endif
-
- ! get hru_dim dimension length (ie., the number of hrus in entire domain)
- err = nf90_inq_dimid(ncid, "nhru", hruDimID);             if(err/=0)then; message=trim(message)//'problem finding nHRU dimension'; return; endif
- err = nf90_inquire_dimension(ncid, hruDimID, len = nHRU); if(err/=0)then; message=trim(message)//'problem reading nHRU dimension'; return; endif
  
- allocate(gru_struc(nGRU), index_map(nHRU), stat=err)
- if(err/=0)then; err=20; message=trim(message)//'problem allocating space for mapping structures'; return; endif
+ if (strtHRU<=0) then
+  ! this is a full run. 
+  ! get hru_dim dimension length (ie., the number of hrus in entire domain)
+  err = nf90_inq_dimid(ncid, "hru", hruDimID);             if(err/=0)then; message=trim(message)//'problem finding hru dimension'; return; endif
+  err = nf90_inquire_dimension(ncid, hruDimID, len = nHRU); if(err/=0)then; message=trim(message)//'problem reading hru dimension'; return; endif
  
- allocate(gru_id(nGRU),hru_id(nHRU),hru2gru_id(nHRU),stat=err)
+ end if 
+ ! read the hruID and HRU's gruID
+ allocate(gru_id(nHRU),hru_id(nHRU),hru2gru_id(nHRU),index_map(nHRU),stat=err)
```

Comment:
That is the maximum number of `gru_id` (1:1 HRU: GRU case). Before locating all the unique GRUs, I just declare a variable big enough to store all the unique `gru_id`. It's a local variable.